src/groupBy.js

```diff
@@ -48,7 +47,10 @@ var append = require('./append');
 module.exports = _curry2(_dispatchable('groupBy', _xgroupBy, function groupBy(fn, list) {
   return _reduce(function(acc, elt) {
     var key = fn(elt);
-    acc[key] = append(elt, acc[key] || (acc[key] = []));
+    if (acc[key] == null) {
```

Comment:
@davidchambers if we change this to be defined in terms of "reduceBy" it's not going to be an issue, because it uses "_has", which, IIIRC, looks only on the object's own keys.